source/developers-guide/custom-shopping-world-elements/index.md

```diff
@@ -419,7 +427,7 @@ class VimeoComponentHandler implements ComponentHandlerInterface
 {
     public function supports(Element $element)
     {
-        return $element->getComponent()->getType() === 'emotion-components-vimeo';
+        return $element->getComponent()->getTemplate() === 'emotion_vimeo';
     }
```

Comment:
Please revert this line. The rest is great!

Reply:
But then the code won't work at that point:
I removed the xtype in the createOrUpdate()-method because you don't need it to create the element or for the handler but I mention it when it is needed to create the custom view in the "Adding a custom emotion component in ExtJS" section.
This advanced section is talked about after the handler section, so I changed from type check to template check. You don't have a custom xtype there to check for.

And if you define the xtype in the element at the beginning of the doc you can't say "[...] we already see a full functional shopping world element [...]" in the "Creating a frontend template for the element" section because the element won't work. There is no js for that xtype so the element does not open a new window with the elements settings.

Reply:
@swDennis